src/xcode/ENA/ENA/Source/HTTPClientCore/Service/__tests__/RestServiceProviderStub.swift

```diff
@@ -5,7 +5,16 @@
 #if !RELEASE
 
 struct LoadResource {
-	let result: Result<Any, Error>
+
+	init(
+		result: @escaping @autoclosure () throws -> Result<Any, Error>,
```

Comment:
😳

Reply:
Why is it needed? Tried to find out where it is used, but compiler still compiles with other LoadResource 🤷‍♂️

Reply:
This doesn't change the call side. It just packs whatever is at the call side inside a closure and we can execute it later, only when needed. I needed this in my unit tests because the result of the second request depends on the first already being executed. In the old client implementation I was able to use a closure to define the result of a request, so I went with the `@autoclosure` here, as I was able to keep the same structure for the tests that needed to be refactored and I didn't need to change much on the ones that were already using the `LoadResource`.

Reply:
Could you point me to one of these tests?

Reply:
One example is `testTestCertificateRegistrationAndExecution_Success()` where the `keyPair` for the second LoadResource is only available after the first LoadResource went through. Most of the tests in the same test class do the same thing

Reply:
Ok got it now. Nice solution! Didn't know this trick 🧙‍♂️